File: test_db_connection.py
import psycopg2
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Get the database URL from environment variables
database_url = os.getenv('DATABASE_URL')

# Try with the standard Supabase format instead of db.subdomain
if 'db.' in database_url:
    standard_url = database_url.replace('db.', '')
    print(f"Attempting to connect to standard URL: {standard_url}")
    
    try:
        # Attempt to connect to the database with standard URL
        conn = psycopg2.connect(standard_url)
        print("Successfully connected to the database with standard URL!")
        
        # Create a cursor
        cur = conn.cursor()
        
        # Execute a simple query
        cur.execute("SELECT version();")
        version = cur.fetchone()
        print(f"Database version: {version[0]}")
        
        # Close the cursor and connection
        cur.close()
        conn.close()
        
    except Exception as e:
        print(f"Error connecting to database with standard URL: {e}")

print(f"Attempting to connect to: {database_url}")

try:
    # Attempt to connect to the database
    conn = psycopg2.connect(database_url)
    print("Successfully connected to the database!")
    
    # Create a cursor
    cur = conn.cursor()
    
    # Execute a simple query
    cur.execute("SELECT version();")
    version = cur.fetchone()
    print(f"Database version: {version[0]}")
    
    # Close the cursor and connection
    cur.close()
    conn.close()
    
except Exception as e:
    print(f"Error connecting to database: {e}")

File: test-db-connection.js
#!/usr/bin/env node
/* Test database connection */

import postgres from 'postgres';
import dotenv from 'dotenv';

// Load environment variables
dotenv.config();

async function testConnection() {
  const dbUrl = process.env.DATABASE_URL;
  if (!dbUrl) {
    console.error('DATABASE_URL not set. Please check your .env file.');
    process.exit(1);
  }

  console.log('Database URL:', dbUrl);

  // Create a postgres client
  const sql = postgres(dbUrl, { 
    ssl: { rejectUnauthorized: false },
    idle_timeout: 5,
    max_lifetime: 30,
    connection_timeout: 5
  });

  try {
    console.log('Testing database connection...');
    // Test the connection with a simple query
    const result = await sql`SELECT version()`;
    console.log('✓ Database connection successful!');
    console.log('Database version:', result[0].version);
    await sql.end();
  } catch (e) {
    console.error('✗ Database connection failed:', e.message);
    try { await sql.end(); } catch (_) {}
    process.exit(1);
  }
}

testConnection();